### PR TITLE
Initial scripts to publish electron.d.ts

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Setting up the scripts that would upload a generated `electron.d.ts` file when we reach an MVP state on `electron-definitelytyped`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +32,7 @@
     "build": "python ./script/build.py -c D",
     "clean": "python ./script/clean.py",
     "coverage": "npm run instrument-code-coverage && npm test -- --use-instrumented-asar",
+    "generate-typescript-definition": "electron-definitelytyped --in=out/electron-api.json --out=out/electron.d.ts",
```
Comment (enlight) [inline package.json]: Could this script name be shortened to `generate-dts`?

**Comment by MarshallOfSound** [comment]
Ping @zeke 

Whats the plan for this / what else needs to be updated in the release pipeline 👍

**Comment by zeke** [comment]
@MarshallOfSound this looks good!

The only thing I see missing is a call to the `generate-dts` script. I suppose it could be invoked from  release.py, but the easiest way to tack it on would probably be just adding `&& npm run generate-dts` to the existing `lint` script. It's not really a linting task, but maybe that's okay...

**Comment by zeke** [comment]
Rebased and updated the `cibuild` script. @zcbenz does that seem like the right place to put it?

**Comment by zeke** [comment]
I just rebased and force-pushed this branch. There are some changes to the linter that will need to land before we can ship this:  https://github.com/electron/electron-docs-linter/pull/85

**Comment by zeke** [comment]
Okay, we renamed the repo and module to [`electron-typescript-definitions`](https://github.com/electron/electron-typescript-definitions) and just published a version first of it at 1.0.0

@MarshallOfSound using the current incantation of `electron-typescript-definitions --in=out/electron-api.json --out=out/electron.d.ts`, is there any guarantee that the generated `electron.d.ts` file is valid? If not, we should figure out a way to incorporate a test into this CLI command, so we can fail the build if the file is invalid.

**Comment by MarshallOfSound** [comment]
@zeke We could just move all devDeps of `electron-typescript-definitions` to be prod deps and run the test command after generating the typings 👍

**Comment by zeke** [comment]
> We could just move all devDeps.

Sounds good. I think there are currently also some `dependencies` that could be moved to  `devDependencies` as well.

**Comment by zeke** [comment]
Okay so we're now generating and tslinting `electron.d.ts` as part of the distribution process and Travis and Appveyor are happy.

@kevinsawicki does this setup look right?

@MarshallOfSound are you happy with the current `electron.d.ts` this is generating?

If so, let's ship this!

**Pull request merged**